src/main/java/com/google/googleidentity/test/TestInMemoryUserDetailsServiceProvider.java

```diff
@@ -0,0 +1,53 @@
+/*
+    Copyright 2020 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package com.google.googleidentity.test;
+
+import com.google.common.base.Charsets;
+import com.google.common.hash.Hashing;
+import com.google.googleidentity.user.InMemoryUserDetailsService;
+import com.google.googleidentity.user.UserDetails;
+import com.google.googleidentity.user.UserDetailsService;
+import com.google.inject.Provider;
+
+public class TestInMemoryUserDetailsServiceProvider implements Provider<UserDetailsService> {
+
+    private static final String TESTUSERNAME0 = "user";
+    private static final String TESTUSERPASSWORD0 = "123456";
+    private static final String TESTUSERNAME1 = "user1";
+    private static final String TESTUSERPASSWORD1 = "12345678";
+
+    @Override
+    public UserDetailsService get() {
```

Comment:
Consider wrapping this up in a module and using the @Provides annotation instead of implementing Provider<>, since the module that binds this is pretty small otherwise.

Also, let Guice instantiate InMemoryUserDetailsService for you so that you don't have to update this code if there are any dependencies added in the future.

```
public final class TestUserModule extends AbstractModule {
  @Provides
  @Singleton
  public UserDetailsService provideUserDetailsService(InMemoryUserDetailsService userDetailsService) {
    userDetailsService.addUser(...);
    return userDetailsService;
  }
}
```